> > [axonivy/core#12835](https://github.com/axonivy/core/pull/12835) [axonivy/core#12840](https://github.com/axonivy/core/pull/12840)
> > I could not yet add the RestClientProjectValidatior because of this error. Will look at this to a later point: <img alt="grafik" width="2000" height="221" src="https://private-user-images.githubusercontent.com/141223521/580806212-b05e27a9-142e-4aad-8550-522cd1089abc.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzY2ODkxNTksIm5iZiI6MTc3NjY4ODg1OSwicGF0aCI6Ii8xNDEyMjM1MjEvNTgwODA2MjEyLWIwNWUyN2E5LTE0MmUtNGFhZC04NTUwLTUyMmNkMTA4OWFiYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNDIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDQyMFQxMjQwNTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMTcyYzczNjRmN2NjZDFiZjhhNTE0NmJkN2NkMWQwYzU1MDNjNjgyYThjOWZmYzQ2OTlkMzQwZjg4ZGUwMjNlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.OSXnNJvR7t80_wR3uhNPQ3taR7cawUAw9RKFmMK5KMI">
> 
> what is the root cause? please send stacktraces as text :)

java.lang.NoClassDefFoundError: javax/ws/rs/core/Feature
	at ch.ivyteam.ivy.rest.client.config.impl.FeatureValidator.<clinit>(FeatureValidator.java:23)
	at ch.ivyteam.ivy.rest.client.config.impl.RestClientProjectValidator$Validator.validateFeatures(RestClientProjectValidator.java:138)
	at ch.ivyteam.ivy.rest.client.config.impl.RestClientProjectValidator$Validator.validate(RestClientProjectValidator.java:48)
	at ch.ivyteam.ivy.rest.client.config.impl.RestClientProjectValidator.validate(RestClientProjectValidator.java:28)
	at ch.ivyteam.ivy.maven.compile.ValidateProjectMojo.validateProject(ValidateProjectMojo.java:59)
	at ch.ivyteam.ivy.maven.compile.ValidateProjectMojo.execute(ValidateProjectMojo.java:52)
	at ch.ivyteam.ivy.maven.compile.TestValidateProjectMojo.validate(TestValidateProjectMojo.java:82)
Caused by: java.lang.ClassNotFoundException: javax.ws.rs.core.Feature
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:580)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:490)
	... 7 more

--- a/pom.xml
+++ b/pom.xml
@@ -76,6 +76,12 @@
       <version>6.0.0</version>
     </dependency>
     <dependency>
+      <!-- Required for project validation -->
+      <groupId>javax.ws.rs</groupId>
+      <artifactId>javax.ws.rs-api</artifactId>
+      <version>2.1.1</version>
+    </dependency>
+    <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-plugin-api</artifactId>
       <version>${maven.version}</version>

--- a/src/main/java/ch/ivyteam/ivy/maven/compile/ValidateProjectMojo.java
+++ b/src/main/java/ch/ivyteam/ivy/maven/compile/ValidateProjectMojo.java
@@ -17,8 +17,10 @@ import ch.ivyteam.ivy.project.model.basic.BasicProjectBuilder;
 import ch.ivyteam.ivy.project.validation.ProjectValidator;
 import ch.ivyteam.ivy.project.validation.ProjectValidatorContext;
 import ch.ivyteam.ivy.project.validation.ProjectValidatorResult.Message;
+import ch.ivyteam.ivy.rest.client.config.impl.RestClientProjectValidator;
 import ch.ivyteam.ivy.role.impl.RoleProjectValidator;
 import ch.ivyteam.ivy.user.impl.UserProjectValidator;
+import ch.ivyteam.ivy.vars.impl.VariableProjectValidator;
 import ch.ivyteam.ivy.webservice.datamodel.impl.WebServiceClientProjectValidator;
 
 /**
@@ -67,7 +69,9 @@ public class ValidateProjectMojo extends AbstractMojo {
     return List.of(
         new UserProjectValidator(),
         new RoleProjectValidator(),
-        new WebServiceClientProjectValidator());
+        new WebServiceClientProjectValidator(),
+        new VariableProjectValidator(),
+        new RestClientProjectValidator());
   }
 
   private void log(Message message) {
@@ -93,7 +97,7 @@ public class ValidateProjectMojo extends AbstractMojo {
     }
 
     private ProjectValidatorContext build() {
-      return ProjectValidatorContext.create(toProject(), toAllProjects());
+      return ProjectValidatorContext.create().project(toProject()).allProjects(toAllProjects()).toContext();
     }
 
     private Project toProject() {

--- a/src/test/java/ch/ivyteam/ivy/maven/compile/TestValidateProjectMojo.java
+++ b/src/test/java/ch/ivyteam/ivy/maven/compile/TestValidateProjectMojo.java
@@ -60,14 +60,34 @@ class TestValidateProjectMojo {
       """;
     Files.writeString(file, content);
 
+    file = dir.resolve("rest-clients.yaml");
+    content = """
+      RestClients:
+        test name:
+          Name: Test
+        test.name:
+          Name: Another
+      """;
+    Files.writeString(file, content);
+
+    file = dir.resolve("variables.yaml");
+    content = """
+      Variables:
+        Test: Something
+        Test: Something else
+      """;
+    Files.writeString(file, content);
     var log = new LogCollector();
     mojo.setLog(log);
     mojo.execute();
     assertThat(log.getWarnings().toString())
         .contains("config/users.yaml: User 'Alex' is configured to have role 'Gangster' which is not defined.")
         .contains("config/webservice-clients.yaml: The web service client key 'test.name' should be sanitized to 'testname' to avoid potential issues. Use the name for a better readability.")
-        .contains("config/webservice-clients.yaml: The web service client key 'test name' should be sanitized to 'test-name' to avoid potential issues. Use the name for a better readability.");
+        .contains("config/webservice-clients.yaml: The web service client key 'test name' should be sanitized to 'test-name' to avoid potential issues. Use the name for a better readability.")
+        .contains("config/rest-clients.yaml: The rest client key 'test.name' should be sanitized to 'testname' to avoid potential issues. Use the name for a better readability.")
+        .contains("config/rest-clients.yaml: The rest client key 'test name' should be sanitized to 'test-name' to avoid potential issues. Use the name for a better readability.");
     assertThat(log.getErrors().toString())
-        .contains("config/roles.yaml: Role 'HR Manager' has an unknown parent 'Manager'.");
+        .contains("config/roles.yaml: Role 'HR Manager' has an unknown parent 'Manager'.")
+        .contains("config/variables.yaml: Variable 'Test' is defined multiple times in variables.yaml.");
   }
 }

